src/logging/transports/fileTransport.ts

```diff
@@ -0,0 +1,37 @@
+import { format, transports } from 'winston';
+
+import { Config } from '../../Config';
+import { nodeUtilFormat, stripAnsi, stripTimestamp } from '../transformers';
+
+const { config } = Config;
+
+/**
+ * File transport for winston.Logger.
+ */
+export function fileTransport() {
+	const transformers = [
+		stripTimestamp(),
+		nodeUtilFormat(),
+		format.timestamp({ format: 'YYYY-MM-DD HH:mm:ss.SSS' }),
+		format.prettyPrint(),
+	];
+
+	if (config.FILE_STRIP_ANSI) {
+		console.info('STRIP');
+		// If the user opts in to strip ANSI characters add the appropriate transformer.
+		// IMPORTANT: We need to add this at the front so it is before `prettyPrint`
+		transformers.unshift(stripAnsi());
+	} else {
+		console.debug('NO STRIP');
+	}
+
+	return new transports.File({
+		level: config.FILE_LEVEL || 'http',
+		filename: config.FILE_PATH || `./logs/file-transport.log`,
+		maxsize: config.FILE_SIZE || 524_288_000, // 500MB
+		maxFiles: config.FILE_COUNT || 2, // Max 1gb
```

Comment:
It does not stop logging, it will just delete the oldest file. Should we try to do something else?

Reply:
It deletes the oldest and creates a new file? I'd prefer we don't try to manage log file size at all and just let `logrotate` (or whatever the node operator prefers) to do its thing. Having it here sure is convenient but I'm afraid it'll lead to issues down the line when `sidecar` tries to delete a file while some other process is busy reading/rotating it.

I think it's fine for now and no need to block this PR, but definitely something worth pointing out in the docs.

Reply:
If we take out `maxSize` and `maxFile` then it will just keep writing to the same file. That would be a super quick change. We could also have yet another config option that just says "Only write to one file". Either way is ok for me. 
Honestly the main reason I had maxFile and maxSize is so some user who isn't really aware of the logs has random a random issue where Sidecar is sucking up disk space.

Reply:
> Honestly the main reason I had maxFile and maxSize is so some user who isn't really aware of the logs has random a random issue where Sidecar is sucking up disk space.

The default is to send logs to stdout only yes? If they opt in to log to file but do not read the docs and do not set up log rotation properly... well, that's on them.